server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java

```diff
@@ -150,58 +143,60 @@ public void tearDown() {
+    public void putMailRepositoryShouldReturnOkWhenRepositoryIsCreated() {
         given()
-            .params("protocol", "url")
+            .params("protocol", "memory")
         .when()
             .put(PATH_ESCAPED_MY_REPO)
         .then()
             .statusCode(HttpStatus.NO_CONTENT_204);
 
-        verify(mailRepositoryStore).create(URL_MY_REPO);
-        verifyNoMoreInteractions(mailRepositoryStore);
+        assertThat(mailRepositoryStore.get(URL_MY_REPO))
+            .isNotEmpty()
+            .containsInstanceOf(MemoryMailRepository.class);
     }
 
     @Test
     public void putMailRepositoryShouldReturnOkWhenRepositoryAlreadyExists() throws Exception {
         given()
-            .params("protocol", "url")
+            .params("protocol", "memory")
         .when()
             .put(PATH_ESCAPED_MY_REPO)
         .then()
             .statusCode(HttpStatus.NO_CONTENT_204);
 
         given()
-            .params("protocol", "url")
+            .params("protocol", "memory")
         .when()
             .put(PATH_ESCAPED_MY_REPO)
         .then()
             .statusCode(HttpStatus.NO_CONTENT_204);
 
-        verify(mailRepositoryStore, times(2)).create(URL_MY_REPO);
-        verifyNoMoreInteractions(mailRepositoryStore);
+        assertThat(mailRepositoryStore.get(URL_MY_REPO))
+            .isNotEmpty()
+            .containsInstanceOf(MemoryMailRepository.class);
+
+        assertThat(mailRepositoryStore.getByPath(PATH_MY_REPO))
+            .hasSize(1);
     }
 
     @Test
-    public void putMailRepositoryShouldReturnServerErrorWhenCannotCreateRepository() throws Exception {
-        when(mailRepositoryStore.create(any()))
-            .thenThrow(new MailRepositoryStore.MailRepositoryStoreException("Error while selecting repository url://myRepo"));
-
+    public void putMailRepositoryShouldReturnServerErrorWhenCannotCreateRepository() {
         given()
-            .params("protocol", "url")
+            .params("protocol", "nonSupportProtocol")
         .when()
             .put(PATH_ESCAPED_MY_REPO)
         .then()
             .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
             .body("statusCode", is(500))
             .body("type", is(ErrorResponder.ErrorType.SERVER_ERROR.getType()))
-            .body("message", is("Error while creating a mail repository with path 'myRepo' and protocol 'url'"))
-            .body("details", is("Error while selecting repository url://myRepo"));
+            .body("message", is("Error while creating a mail repository with path 'myRepo' and protocol 'nonSupportProtocol'"))
+            .body("details", is("No Mail Repository associated with nonSupportProtocol"));
     }
 
     @Test
     public void getMailRepositoriesShouldReturnEmptyWhenEmpty() {
-        when(mailRepositoryStore.getPaths()).thenAnswer(RETURN_NO_MAIL_REPOSITORY);
+        mailRepositoryStore = new MailRepositoryStoreImpl(new MemoryMailRepositoryUrlStore(), Sets.newHashSet());
```

Comment:
Is the reallocation needed? Isn't the other mailRepoStore also empty?

BTW what is `Sets.newHashSet()` ?

Reply:
you're right, this statement is useless. `Sets.newHashSet()` is the set of `MailRepositoryProvider`